set.mm

```diff
@@ -613950,6 +613958,31 @@ number axioms (add ~ ax-10 , ~ ax-11 , ~ ax-13 , ~ ax-nul , and remove
   1t1e1ALT $p |- ( 1 x. 1 ) = 1 $=
     ( c1 cr wcel cmul co wceq 1re ax-1rid ax-mp ) ABCAADEAFGAHI $.
 
+  ${
+    $d x y A $.  $d x y B $.
+    $( Addition with 1 is commutative for natural numbers.  (Contributed by
+       Steven Nguyen, 9-Dec-2022.) $)
+    nnadd1com $p |- ( A e. NN -> ( A + 1 ) = ( 1 + A ) ) $=
+      ( vx vy cv c1 caddc co wceq oveq1 oveq2 eqeq12d eqid cn wcel 1cnd addassd
+      weq nncn sylan9eqr ex nnind ) BDZEFGZEUBFGZHEEFGZUEHCDZEFGZEUFFGZHZUGEFGZ
+      EUGFGZHZAEFGZEAFGZHBCAUBEHUCUEUDUEUBEEFIUBEEFJKBCQUCUGUDUHUBUFEFIUBUFEFJK
+      UBUGHUCUJUDUKUBUGEFIUBUGEFJKUBAHUCUMUDUNUBAEFIUBAEFJKUELUFMNZUIULUIUOUJUH
+      EFGUKUGUHEFIUOEUFEUOOZUFRUPPSTUA $.
+
+    $( Addition is commutative for natural numbers.  Uses fewer axioms than
+       ~ addcom .  (Contributed by Steven Nguyen, 9-Dec-2022.) $)
+    nnaddcom $p |- ( ( A e. NN /\ B e. NN ) -> ( A + B ) = ( B + A ) ) $=
```

Comment:
Can you do these for NN0, or again is there a problem related to the current definition of NN0 ? If so, we may want to revise the definitions, and define NN0 directly and then NN from it.

Reply:
yeah using 0 would basically be a longer version of nncn and addcom as far as I can tell

---
pretty much all theorems involving addition and just the digits 1 through 9 are really just ax-1cn, ax-addcl, and ax-addass:

4p5e9 ( 4 + 5 ) = 9
expands to
( ( ( ( 1 + 1 ) + 1 ) + 1 ) + ( ( ( ( 1 + 1 ) + 1 ) + 1 ) + 1 ) ) =
( ( ( ( ( ( ( ( 1 + 1 ) + 1 ) + 1 ) + 1 ) + 1 ) + 1 ) + 1 ) + 1 )

With df-dec and/or multiplication, mulid1 and ax-distr is needed
;11 = ( ( ( 9 + 1 ) x. 1 ) + 1 ) which by mulid1 eventually becomes ( 9 + 2 ) [9p2e11 6p5e11 7p5e12 et al will be improved in the future like this]

;21 is the same thing but ax-distr to expand the multiplication

( 7 x. 3 ) expands with ax-distr into ( ( 7 x. 1 ) + ( ( 7 x. 1 ) + ( 7 x. 1 ) ) )

But the mere addid1 uses every complex number axiom except cnex and supremum...

Reply:
Ok. So changing the definition of NN0 wouldn't change that, correct ? Maybe you can add some of this in a comment, if you find a good phrasing and location to put it.